### ISSUE TITLE
ToDo: diffs FF53-FF54

### ISSUE THREAD

**Issue by earthlng** [ghacksuserjs/ghacks-user.js#87]
[**v53.0 vs v54.0**](https://github.com/earthlng/FFprefs-diffs/blob/master/diffs/5x/diff-v53.0-vs-v54.0.log.js)

119 diffs ( 54 new, 55 gone, 10 different )

## new in v54.0:
* [x] pref("browser.formautofill.enabled", false); https://github.com/ghacksuserjs/ghacks-user.js/commit/ec03969d98f3f8f26ba232aa772a5bb0ca4c8e63
* [x] pref("browser.newtabpage.activity-stream.enabled", false); https://github.com/ghacksuserjs/ghacks-user.js/commit/1621cd5e4805a9ad47536bce207945ab27fee643
* [x] pref("browser.urlbar.usepreloadedtopurls.enabled", false); https://github.com/ghacksuserjs/ghacks-user.js/commit/1a04c1314e47ffa31fd7a3ef5fd5ee906784324a
* [x] MistakeURLs: https://github.com/ghacksuserjs/ghacks-user.js/commit/d3e1fe3a1c5d91e73dda224810806adcceed0d8a
   * https://hg.mozilla.org/mozilla-central/rev/077f42a9964c#l10.1
   * pref("browser.safebrowsing.provider.google.reportMalwareMistakeURL", "https://%LOCALE%.malware-error.mozilla.com/?hl=%LOCALE%&url=");
   * pref("browser.safebrowsing.provider.google.reportPhishMistakeURL", "https://%LOCALE%.phish-error.mozilla.com/?hl=%LOCALE%&url=");
   * pref("browser.safebrowsing.provider.google4.reportMalwareMistakeURL", "https://%LOCALE%.malware-error.mozilla.com/?hl=%LOCALE%&url=");
   * pref("browser.safebrowsing.provider.google4.reportPhishMistakeURL", "https://%LOCALE%.phish-error.mozilla.com/?hl=%LOCALE%&url=");
* [x] pref("dom.ipc.processCount.extension", 1); https://github.com/ghacksuserjs/ghacks-user.js/commit/dd01dd54c65de3c62ee1e91666ed3dedec44b95b
* [x] pref("extensions.screenshots.system-disabled", true); - https://github.com/ghacksuserjs/ghacks-user.js/commit/f61c951ca0212a9f1b0b16a86ab5c222f0889a9b
* [x] pref("geo.security.allowinsecure", true); https://github.com/ghacksuserjs/ghacks-user.js/commit/f3a0e8d88c109dc6b0ab44e5a1e440b306408b56
* [x] pref("network.http.referer.hideOnionSource", false); https://github.com/ghacksuserjs/ghacks-user.js/commit/ffd89807bf9a247de48cdd226da78a92da0ee778
* [x] pref("privacy.firstparty.isolate.restrict_opener_access", true); https://github.com/ghacksuserjs/ghacks-user.js/commit/da73ca1759a0445526c656571b80a71b75660f56
* [x] pref("security.sandbox.logging.enabled", false); https://github.com/ghacksuserjs/ghacks-user.js/commit/dd01dd54c65de3c62ee1e91666ed3dedec44b95b
* pref("security.data_uri.inherit_security_context", true);
   * added to "items to keep an eye on: issue 20

```js
pref("intl.tsf.associate_imc_only_when_imm_ime_is_active", false);
pref("intl.tsf.hack.ms_japanese_ime.do_not_associate_imc_on_win10", true);
```

## removed, renamed or hidden in v54.0:
Done - see https://github.com/ghacksuserjs/ghacks-user.js/commit/82432a3d402f0349312caad3bf8776cdfac4d05a
* [x] pref("browser.safebrowsing.reportMalwareMistakeURL", "https://%LOCALE%.malware-error.mozilla.com/?hl=%LOCALE%&url="); https://hg.mozilla.org/mozilla-central/rev/077f42a9964c#l10.1
* [x] pref("browser.safebrowsing.reportPhishMistakeURL", "https://%LOCALE%.phish-error.mozilla.com/?hl=%LOCALE%&url="); https://hg.mozilla.org/mozilla-central/rev/077f42a9964c#l10.1
* [x] pref("dom.archivereader.enabled", false); [~~1342361~~](https://bugzilla.mozilla.org/show_bug.cgi?id=1342361 "RESOLVED FIXED in Firefox 54 - Get rid of ArchiveReader API")
* [x] pref("media.eme.apiVisible", true); [~~1242321~~](https://bugzilla.mozilla.org/show_bug.cgi?id=1242321 'RESOLVED FIXED in Firefox 54 - Remove obsolete "media.eme.apiVisible" pref')

## changed in v54.0:
* [x] `pref("extensions.blocklist.url", "https://blocklists.settings.services.mozilla.com/v1/blocklist/3/%APP_ID%/%APP_VERSION%/%PRODUCT%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/%PING_COUNT%/%TOTAL_PING_COUNT%/%DAYS_SINCE_LAST_PING%/"); // prev: "https://blocklist.addons.mozilla.org/blocklist/3/%APP_ID%/%APP_VERSION%/%PRODUCT%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/%PING_COUNT%/%TOTAL_PING_COUNT%/%DAYS_SINCE_LAST_PING%/"`
=> [Migrate the blocklist preferences to new URI](https://bugzilla.mozilla.org/show_bug.cgi?id=1341332)
* [x] pref("media.peerconnection.ice.tcp", true); // prev: false - https://github.com/ghacksuserjs/ghacks-user.js/commit/ab7dfb7f43770583eebaa5b604e065d43fb4286a

## ignore
==NEW
```js
pref("apz.allow_with_webrender", false);
pref("browser.formautofill.loglevel", "Warn"); // see master switch to disable
pref("browser.urlbar.usepreloadedtopurls.expire_days", 14); // see master switch to disable
pref("browser.storageManager.pressureNotification.minIntervalMS", 1200000);
pref("browser.storageManager.pressureNotification.usageThresholdGB", 5);
pref("devtools.gridinspector.showGridOutline", false);
pref("devtools.source-map.locations.enabled", false);
pref("dom.forms.select.customstyling", true); // https://bugzilla.mozilla.org/show_bug.cgi?id=1339966
pref("dom.ipc.plugins.forcedirect.enabled", true); // we do not support flash
pref("dom.moduleScripts.enabled", false); // ECMAScripts
pref("dom.timeout.max_consecutive_callbacks", 5);
pref("dom.vr.puppet.enabled", false);
pref("dom.vr.test.enabled", false);
pref("extensions.e10sMultiBlocksEnabling", true); // leave e10s experimental staging alone
pref("gfx.downloadable_fonts.keep_variation_tables", false);
pref("gfx.downloadable_fonts.otl_validation", false);
pref("gfx.webrender.enabled", true);
pref("image.mem.animated.discardable", false);
pref("image.mem.shared", false);
   // ^^ https://bugzilla.mozilla.org/show_bug.cgi?id=1331944
   // ^^ https://bugzilla.mozilla.org/show_bug.cgi?id=1339202
pref("intl.uidirection", -1);
pref("layers.advanced.border-layers", false);
pref("layers.advanced.caret-layers", false);
pref("layout.css.text-justify.enabled", false);
pref("layout.scrollbars.always-layerize-track", false);
/* https://developer.mozilla.org/en-US/docs/Mozilla/QA/Marionette ***/
pref("marionette.enabled", false);
pref("marionette.forcelocal", true);
pref("marionette.log.level", "info");
pref("marionette.port", 2828);
pref("marionette.prefs.recommended", true);
pref("media.eme.vp9-in-mp4.enabled", false);
pref("media.ffmpeg.low-latency.enabled", false);
pref("services.sync.engine.bookmarks.validation.interval", 86400);
pref("services.sync.engine.bookmarks.validation.maxRecords", 1000);
pref("services.sync.engine.bookmarks.validation.percentageChance", 10);
pref("signon.masterPasswordReprompt.timeout_ms", 900000);
pref("network.throttle.enable", true);
pref("network.throttle.resume-for", 2000);
pref("network.throttle.suspend-for", 2000);
```

==REMOVED or HIDDEN
```js
pref("browser.newtabpage.remote", false);
pref("browser.newtabpage.remote.content-signing-test", false);
pref("browser.newtabpage.remote.keys", "");
pref("browser.newtabpage.remote.mode", "production");
pref("browser.newtabpage.remote.version", "1");
pref("devtools.sourcemap.locations.enabled", false);
pref("devtools.toolbox.toolbarSpec", '["splitconsole", "paintflashing toggle","scratchpad","resize toggle","screenshot --fullpage --file", "rulers", "measure"]');
pref("gfx.font_rendering.cleartype.always_use_for_content", false);
pref("gfx.font_rendering.cleartype.use_for_downloadable_fonts", true);
pref("intl.uidirection.ar", "rtl");
pref("intl.uidirection.fa", "rtl");
pref("intl.uidirection.he", "rtl");
pref("intl.uidirection.ug", "rtl");
pref("intl.uidirection.ur", "rtl");
// https://bugzilla.mozilla.org/show_bug.cgi?id=1343941
pref("layers.allow-d3d9-fallback", false);
pref("layers.prefer-d3d9", false);
pref("layout.css.object-fit-and-position.enabled", true);
pref("layout.css.unprefixing-service.enabled", true);
// remove HTTP1 Pipeline Support: https://bugzilla.mozilla.org/show_bug.cgi?id=1340655
pref("network.http.pipelining", false);
pref("network.http.pipelining.abtest", false);
pref("network.http.pipelining.aggressive", false);
pref("network.http.pipelining.max-optimistic-requests", 4);
pref("network.http.pipelining.maxrequests", 32);
pref("network.http.pipelining.maxsize", 300000);
pref("network.http.pipelining.read-timeout", 30000);
pref("network.http.pipelining.reschedule-on-timeout", true);
pref("network.http.pipelining.reschedule-timeout", 1500);
pref("network.http.pipelining.ssl", false);
pref("network.http.proxy.pipelining", false);
pref("pointer-lock-api.prefixed.enabled", false);
pref("security.sandbox.windows.log", false);
// https://bugzilla.mozilla.org/show_bug.cgi?id=1296767#c125
pref("services.sync.jpake.firstMsgMaxTries", 300);
pref("services.sync.jpake.lastMsgMaxTries", 300);
pref("services.sync.jpake.maxTries", 10);
pref("services.sync.jpake.pollInterval", 1000);
pref("services.sync.jpake.serverURL", "https://setup.services.mozilla.com/");
pref("services.sync.log.logger.service.jpakeclient", "Debug");
pref("services.sync.log.logger.userapi", "Debug");
pref("services.sync.miscURL", "misc/");
pref("services.sync.privacyURL", "https://services.mozilla.com/privacy-policy/");
pref("services.sync.scheduler.sync11.singleDeviceInterval", 86400);
pref("services.sync.serverURL", "https://auth.services.mozilla.com/");
pref("services.sync.statusURL", "https://services.mozilla.com/status/");
pref("services.sync.syncKeyHelpURL", "https://services.mozilla.com/help/synckey");
pref("services.sync.termsURL", "https://services.mozilla.com/tos/");
pref("services.sync.userURL", "user/");
// ---------------------------------------------------------------------------------------
// renamed to 'services.sync.engine.bookmarks.validation.*'
pref("services.sync.validation.interval", 86400);
pref("services.sync.validation.maxRecords", 100);
pref("services.sync.validation.percentageChance", 10);
// https://bugzilla.mozilla.org/show_bug.cgi?id=1313045
pref("toolkit.identity.debug", false);
pref("toolkit.identity.enabled", false);
```

==CHANGED
```js
pref("browser.safebrowsing.provider.google4.lists", "goog-badbinurl-proto,goog-downloadwhite-proto,goog-phish-proto,googpub-phish-proto,goog-malware-proto,goog-unwanted-proto"); // prev: "goog-phish-proto,googpub-phish-proto,goog-malware-proto,goog-unwanted-proto"
pref("dom.event.highrestimestamp.enabled", true); // prev: false
pref("dom.ipc.plugins.asyncdrawing.enabled", true); // prev: false // we do not support flash
pref("extensions.blocklist.detailsURL", "https://blocked.cdn.mozilla.net/"); // prev: "https://www.mozilla.com/%LOCALE%/blocklist/"
pref("extensions.blocklist.itemURL", "https://blocked.cdn.mozilla.net/%blockID%.html"); // prev: "https://blocklist.addons.mozilla.org/%LOCALE%/%APP%/blocked/%blockID%"
pref("font.name-list.serif.x-math", "..."); // removed 'Symbol' from the list
pref("layout.css.clip-path-shapes.enabled", true); // prev: false
pref("security.mixed_content.hsts_priming_cache_timeout", 604800); // prev: 10080
```

**Comment by earthlng** [comment]
`javascript.options.shared_memory` - https://github.com/ghacksuserjs/ghacks-user.js/issues/47#issuecomment-287331831 + https://github.com/ghacksuserjs/ghacks-user.js/issues/47#issuecomment-287392688

**Comment by crssi** [comment]
Nice work @earthlng.

Those looks promising for "lite" version for FF 54+:
`browser.safebrowsing.provider.google.reportMalwareMistakeURL`
`browser.safebrowsing.provider.google.reportPhishMistakeURL`
`browser.safebrowsing.provider.google4.reportMalwareMistakeURL`
`browser.safebrowsing.provider.google4.reportPhishMistakeURL`

Assuming that I need to trust some entity, switching those to mozilla servers instead of google is a good thing... at least I also have current (the google ones) to false now.

**Comment by earthlng** [comment]
Thanks @crssi

In a "lite" version we would probably enable safebrowsing and trackingprotection and therefore not change those 4 prefs, but since we don't have different versions at the moment and since we have the 2 "old" prefs already set to empty string (browser.safebrowsing.reportMalwareMistakeURL + browser.safebrowsing.reportPhishMistakeURL), we will most likely also set the 4 new prefs to empty string in the master user.js.

**Comment by earthlng** [comment]
`extensions.blocklist.url`

```
FF54:    https://blocklists.settings.services.mozilla.com/v1/blocklist/3/ %APP_ID%/%APP_VERSION%/%PRODUCT%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/%PING_COUNT%/%TOTAL_PING_COUNT%/%DAYS_SINCE_LAST_PING%/
FF53:    https://blocklist.addons.mozilla.org/blocklist/3/                %APP_ID%/%APP_VERSION%/%PRODUCT%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/%PING_COUNT%/%TOTAL_PING_COUNT%/%DAYS_SINCE_LAST_PING%/
user.js: https://blocklist.addons.mozilla.org/blocklist/3/                %APP_ID%/%APP_VERSION%/
```

**Comment by crssi** [comment]
I understand... we are on exactly "on the same page" here. :)
Enable for the lite version and "empty string"/disable on the current.

**Comment by Thorin-Oakenpants** [comment]
extensions.e10sMultiBlocksEnabling. We can probably ignore this totally as 1) its just an experiment 2) will be legacy in a few versions even if it makes it to stable 3) we should have experiments covers 0340+0341 and 4) its e10s stuff which we're kinda leaving alone alone anyway
 - https://bugzilla.mozilla.org/show_bug.cgi?id=1349363
 - part of an experiment / tracking ? read the bug
 - probably only used in beta 54 and will be removed from stable?

**Comment by earthlng** [comment]
> probably only used in beta 54 and will be removed from stable?

yeah, maybe. let's ignore it for now and wait until stable is released and then we can see if the pref is still there

**Comment by Radagast** [comment]
If you don't mind my asking, what do you use for calculating the differences between releases. I use Linux and would lie to keep a track of the pref changes on this platform.

Thanks

**Comment by earthlng** [comment]
@Radagast, I don't mind at all. I wrote 2 scripts - one that extracts the omni.ja files from the setup files and then extracts all the preferences files from those 2 omni.ja's. The 2nd script parses those files (in the correct order!) and adds all active prefs to an array (checking if a pref is already in the array and if so update the value) for 2 selected versions. Resulting in a single multi-dimensional array with the pref names in column 0 and the values for the two versions in columns 1 and 2. Then looping over the array comparing the 2 values and outputting it in the correct category (new, changed, missing). Because the script needs to loop over nearly 3000 prefs multiple times it takes around 21 seconds to finish.
If there's a demand for it, I can create and upload the diffs for the linux versions as well in my repo.

**Comment by Thorin-Oakenpants** [comment]
@Radagast @earthlng Keeping linux diff logs at https://github.com/earthlng/FFprefs-diffs would be good. Also ESR (but just the final ones - stable-vs-stable), especially as 57 gets nearer (it's also good to be able to refer to them to find when a pref was first introduced). Great repo there, just needs a little cow bell. :)

**Comment by Radagast** [comment]
> If there's a demand for it, I can create and upload the diffs for the linux versions as well

Thanks for the information. I for one wold be very interested in seeing the Linux prefs added but I don't wish to present you with additional work if the demand is low.

**Comment by earthlng** [comment]
> Also ESR (but just the final ones - stable-vs-stable)

they are all stable afaik, what do you mean? Tell me which versions you want compared and I can do it.
45.0 vs 52.0? That will be a massive list.

> I for one would be very interested in seeing the Linux prefs added

@Radagast, [this](https://github.com/ghacksuserjs/ghacks-user.js/issues/101) will have to do for now, sorry

**Comment by Thorin-Oakenpants** [comment]
re ESR: I don't use ESR, I just assumed they had betas - my bad. I just thought that down the track, when we create an ESR branch (because lets face it, 57 is going to cause a major split), it would be good to be able to refer back to when prefs were introduced/changed/removed - and that if it's no problem (not sure how much work it is for you), to create diffs for ESR from here on in at your repo. Up to you :)

**Comment by earthlng** [comment]
Well, in ESR not too many things change between the minor versions (fe. 52.0 to 52.1.0)

```js
/******************************************************************************************
 * diffs between FF prefs v52.0esr and v52.1.0esr
 ******************************************************************************************/
// changed in v52.1.0esr:
pref("network.predictor.enable-prefetch", false); // prev: true
pref("security.pki.certificate_transparency.mode", 0); // prev: 1
```

At some point there will be massive changes where all the old stuff gets removed and the new stuff added, but atm I'm not sure when exactly that is normally happening.

**Comment by publicarray** [comment]
Does anyone know what `browser.migrate.automigrate.enabled` does?

**Comment by Thorin-Oakenpants** [comment]
`browser.migrate.automigrate.enabled` - landed in 49=>false, was flipped to default true in 53beta but backtracked to false. Looks like 54 they want to enable in. Am digging into it.

https://bugzilla.mozilla.org/show_bug.cgi?id=1297479

godamn funnelcake

**Comment by Thorin-Oakenpants** [comment]
@fmarier et al .. what exactly is funnelcake. I have seen this before, and it was to do (from memory) with a new home page and tiles and news snippets etc.

**Comment by publicarray** [comment]
Thanks for looking into it. I did not expect you take a deep dive. I only thought to ask because at some stage it was added to the [user.js before the upload to github](https://github.com/ghacksuserjs/ghacks-user.js/blob/1778d74bcbf2a9cbc1f43ae61c29c1fe69d2e6f8/user.js#L1396). 

Look what I found: https://bugzilla.mozilla.org/show_bug.cgi?id=1322718#c1
(at the top is a link to a google doc)
PS Thanks for the invite. I can't promise much involvement though.

**Comment by Thorin-Oakenpants** [comment]
@publicarray No worries .. it makes it easy to find you with the @ symbol

Great link - so funnelcake is this new fangled thing about "first-run onboarding experience" and involves tiles, top sites, activity stream etc .. the pref is a UI switch only it seems, but I'm sure the user base here is all about blocking funnelcake

**Comment by publicarray** [comment]
Looks like yes. I looked at https://bugzilla.mozilla.org/show_bug.cgi?id=1297479 and It looks like `browser.migrate.automigrate.enabled` is about auto migrating browser settings from another browser e.g. from Internet Explorer to Firefox. Funnelcake is probably revisiting these prefs and "experiences".

**Comment by Thorin-Oakenpants** [comment]
I took some of the related bugs linked in 1322718 and came across things like the tour and more

Activity stream stuff: https://www.ghacks.net/2017/04/09/firefox-57-photon-mockups-activity-stream-library-compact-mode-more/
Intro tour thingy - https://www.ghacks.net/2017/03/31/firefox-photon-new-design-mockups-show-interface-and-more/

I'm thinking `browser.migrate.automigrate.enabled` can be removed from the js. It's user initiated and there will be other prefs in time to actually control funnelcake bits and bobs if ever needed

**Comment by earthlng** [comment]
If **It's user initiated** why is it called automigrate? It's in the Personal Settings section and I'd like to keep it there and not remove it

**Comment by Thorin-Oakenpants** [comment]
Here's the whole lot: https://dxr.mozilla.org/mozilla-central/source/browser/components/migration/AutoMigrate.jsm
Here's a test: https://dxr.mozilla.org/mozilla-central/source/browser/components/migration/tests/browser/browser_undo_notification_multiple_dismissal.js#7

My understanding is that it needs to be enabled for the undo action to be available. It's hard to find much about this, so some of this is guesswork. This is currently an experiment? The pref is primarily for the experiment (which doesn't make sense to me)? The auto-importing is for new installs/new profiles only? Maybe the pref is both, at false it never imports and therefore the ui for undo/accept is never needed?

Probably landing separately (but still under "funnelweb" I think) is Activity Streams, which will be a system addon ( see [1304181(https://bugzilla.mozilla.org/show_bug.cgi?id=1340181) comment 1 ). I was looking at `browser.newtabpage.activity-stream.enabled` and thinking we should add that now and enforce to default. Am still investigating. The system addon will enforce newtab to use an activity stream url

**Comment by earthlng** [comment]
> Activity Streams will be a system addon

Hallelujah! Praise the lord!! Our prayers have been heard. Finally they stop hard-coding that kinda shit into FF. Thank you Jesus, I knew you didn't die in vain

**Comment by Thorin-Oakenpants** [comment]
^^ except I would like system addons to be listed in about:addons, and enabled/disabled from there rather than pissing around in with removing files

**Comment by earthlng** [comment]
^^ 1st world problems

**Comment by earthlng** [comment]
`network.throttle.*` - IMO we can ignore those. Throttling downloads when a page load happens seems nice. Also no privacy/security implications afaik. [1312754 changeset](https://hg.mozilla.org/mozilla-central/rev/145410f30206)

**Comment by earthlng** [comment]
* `browser.urlbar.usepreloadedtopurls.enabled` - [1211726](https://bugzilla.mozilla.org/show_bug.cgi?id=1211726) = pre-seeding the awesome bar suggestions with a [hardcoded list of URLs](https://hg.mozilla.org/mozilla-central/rev/9762f34b30a4) - I don't think we ever want this so lets add it to the user.js and set it to false. Nightly already sets it to true atm. IMO we can ignore the second pref 
`browser.urlbar.usepreloadedtopurls.expire_days`
* `dom.vr.puppet.enabled` + `dom.vr.test.enabled` - both default to false in FF54 and IMO we can ignore these for now
* `security.sandbox.logging.enabled` - would be nice to have in the PERSONAL section IMO, maybe commented out and titled as "enable sandbox logging"
* `browser.storageManager.pressureNotification.*` - we have `browser.storageManager.enabled;false` in `2706` so IMO we can ignore these 2 new prefs

**Comment by Atavic** [comment]
@earthlng A note here, as I modify any child entry even if the main ones has been modified, e.g. `browser.urlbar.usepreloadedtopurls.expire_days` and `browser.urlbar.usepreloadedtopurls.enabled`
This way, if the latter is modified in some unexpected way, I still have the former one with my settings (like 999 days).

**Comment by earthlng** [comment]
expire_days = 0 (?) or 1 is what you want in that case btw rather than 999 ;)

> as I modify any child entry even if the main ones has been modified

yes, I saw that in your fork :) but we would end up with hundreds of additional prefs if we did that. We want to make this user.js easier to use and adding dozens or even hundreds of additional "unnecessary" prefs would be the opposite of that

**Comment by Atavic** [comment]
I'm ok with different views and just had to tell my own. 

999 days for neutralizing that feature I do not need, see [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1211726#c61).

**Comment by earthlng** [comment]
well, "if the latter is modified in some unexpected way" then setting the expire_days to 999 will show you those pre-seeded urls for 999 days instead of the default 14 days, or (and that's why I said it) 0 or 1 days if you also want to add expire_days in your user.js, which I assume is what you want.

**Comment by earthlng** [comment]
999 days does not neutralize that feature you do not need but rather extends it by quite a bit :)

**Comment by Atavic** [comment]
My ideas: 
- I do not use that, I have a home.html or some pages loaded as command line arguments on launch.
- I'd like to leave that list immaculate, if ever the list got populated, I'd leave as is for some long times.

**Comment by earthlng** [comment]
changes in beta11 since the beta1 diff in OP:

### NEW
pref("extensions.screenshots.system-disabled", true);
~~pref("layout.css.appearance.enabled", true);~~
~~pref("layout.css.moz-appearance.enabled", true);~~
pref("signon.masterPasswordReprompt.timeout_ms", 900000);

### CHANGED
~~pref("browser.crashReports.unsubmittedCheck.enabled", true); // prev: false~~
~~pref("browser.migrate.automigrate.enabled", true); // prev: false~~
pref("font.name-list.serif.x-math", "..."); // removed 'Symbol' from the font list in beta11
~~pref("javascript.options.shared_memory", true); // prev: false~~
~~pref("print.use_simplify_page", true); // prev: false~~
pref("security.mixed_content.hsts_priming_cache_timeout", 604800); // prev: 10080

**Comment by earthlng** [comment]
* `("image.mem.shared", false)` - you added a link to [1331944](https://bugzilla.mozilla.org/show_bug.cgi?id=1331944) but the commit was done in [1339202](https://bugzilla.mozilla.org/show_bug.cgi?id=1339202)

They had to "Revert back to using a pref for the moment" because the older patch (without a pref?) caused a [crash](https://bugzilla.mozilla.org/show_bug.cgi?id=1339202#c4) and had to be backed out.
I'm really not qualified to comment on this and we can only hope that the devs identify and fix as many potential flaws as possible before they enable this pref.
The same goes for `image.mem.animated.discardable` @ https://bugzilla.mozilla.org/show_bug.cgi?id=1343341

* `("gfx.downloadable_fonts.otl_validation", false)` - this is interesting. [1331737](https://bugzilla.mozilla.org/show_bug.cgi?id=1331737)

> Expose a pref to control the validation of OpenType Layout tables, so that Nightly/Aurora users can choose to bypass validation (like we do on Beta/Release) if they really want out-of-spec fonts to be loaded.

> The default behavior will still be as now -- such fonts will load on Beta/Release, **so we don't annoy too many users**, but Nightly/DevEd will block them and issue console error messages to encourage people to report and fix out-of-spec fonts and font-deployment workflows.

I think the Nightly behavior is preferable because allowing "out-of-spec" data parsing is probably never a good thing other than not to "annoy too many users". They also mention that

> there are moves afoot to perhaps modify the spec to allow it, but as of now it's an OpenType spec violation and causes OTS to complain

But they also [say](https://bugzilla.mozilla.org/show_bug.cgi?id=1331737#c2)

> on beta/release, we allow those tables to bypass OTS even if they have spec violations, as **in general harfbuzz can still handle them safely.**

"in general" - well okay - maybe worth considering adding this and set to to "true"

* `("gfx.downloadable_fonts.keep_variation_tables", false)` - [1341085](https://bugzilla.mozilla.org/show_bug.cgi?id=1341085)
similar to the one above this one also allows to "weaken" the OTS sanitizer but it's only made available for testing purposes and I think we can ignore this because "false" is fine

* `("extensions.screenshots.system-disabled", true)` - [Firefox Screenshots integrated in Firefox Nightly](https://www.ghacks.net/2017/05/28/firefox-screenshots-integrated-in-firefox-nightly/)
some users will probably not want this so we should consider adding this, maybe in the Personal section idk

* `security.mixed_content.hsts_priming_cache_timeout` - we have HSTS Priming disabled so this is irrelevant

* `signon.masterPasswordReprompt.timeout_ms` - a 15min timeout for masterPasswordReprompt? no idea what this does but since I don't store passwords in FF maybe you can test this once stable is released and decide if we need to add this and maybe want to change the value.

**Comment by crssi** [comment]
You can store passwords in FF. This store can be configured that can be accessed only with master password.
So 15 min would be that when entered master password for passwords access, you will not be asked any more for 15 min period.

**Comment by earthlng** [comment]
I don't think so. What you are describing is done with the following 2 prefs, isn't it?
```js
/* 0903: set how often Firefox should ask for the master password
 * 0=the first time (default), 1=every time it's needed, 2=every n minutes (as per the next pref) ***/
user_pref("security.ask_for_password", 2);
/* 0904: set how often in minutes Firefox should ask for the master password (see pref above)
 * in minutes, default is 30 ***/
user_pref("security.password_lifetime", 5);
```
I could be wrong of course

**Comment by crssi** [comment]
Its more likely that I am wrong here.

**Comment by earthlng** [comment]
`signon.masterPasswordReprompt.timeout_ms`

[Bug 1348791](https://bugzilla.mozilla.org/show_bug.cgi?id=1348791) - Add a timeout to master password prompt when searching logins

> The login manager searching logins for autofill may trap the user
> in an infinite loop of master password prompts until the user enters
> the correct master password. To prevent that, we're adding a timeout
> to showing the master password prompt for autofill after it was last
> cancelled.

**Comment by earthlng** [comment]
> YAGNI

IMO every pref in the user.js with the same value as the default value in FF should be active.
Otherwise it's confusing because IMO commented prefs are optional hardening stuff that are only commented out because they break too much sites. Therefore for `1202` I would activate the 2nd + 3rd pref **IF** `security.tls.version.max;4` lands in stable. If not then we can wait with the change until it does land.

**Comment by publicarray** [comment]
> IMO every pref in the user.js with the same value as the default value in FF should be active.

@earthlng while I get your point I disagree. By not forcing every default pref we allow the user to change some settings via the FF GUI granted some could be set to less private/secure by the user. Also I'm lazy so if I don't like a pref I don't need to change as many :stuck_out_tongue:. For me I like  that can look at the user.js and I know that all the used prefs that are in use differ from the FF default (i.e. I know what improvements are made through the user.js)

**Comment by Thorin-Oakenpants** [comment]
> For me I like that can look at the user.js and I know that all the used prefs that are in use differ from the FF default

That was never the intention. "enforcing" defaults has always been what we do, in general. If a pref has any security/FP/privacy/tracking etc implications, then it's in the user.js. Some items are clearly noted as FYI and "do not change" (eg e10s/processes stuff and blocking IP6), but for the most part, inactive items (exclude the personal section) are ones we feel break too much or are optional (eg TP/SB)

One reason I've pushed back against this particular pref (TLS versions), is that it wasn't ready. However, we have always (almost) enforced defaults as a "best practice", thus those that are inactive are inactive for a "reason" - so despite YAGNI, maybe we should

I think I'll revisit this in a new issue. There are 86 matches for "// user_" (up to 2800 inclusive) - we should check them all for consistency - eg why? are tags applied [warning], [setup] etc? are [notes] included? etc

**Comment by earthlng** [comment]
`security.data_uri.inherit_security_context` - this is not ready to use just yet. ATM none of mozilla's testfiles can handle this and the code behind it is completely untested. The pref was also already renamed for FF55 btw. IMO we should add this pref to the "to keep an eye on" list, with a link to [1324406](https://bugzilla.mozilla.org/show_bug.cgi?id=1324406 "ASSIGNED - Treat 'data:' documents as unique, opaque origins") which is the main ticket for the problem they're trying to fix with this pref. FYI 1324406 blocks the restricted 1296976 ("Access Denied")
The code most likely works but there are serious concerns that it could open more holes/problems than it tries to solve!
The 2nd link you added above for this pref is for a similar problem ie blocking **top level** window data: URIs, but is not directly related to this pref - one is about not inheriting security context and the other is for blocking top level data:URIs

* [1324406](https://bugzilla.mozilla.org/show_bug.cgi?id=1324406) - Treat 'data:' documents as unique, opaque origins (ASSIGNED)
  * Depends on: 1302399, 1337268, 1337269, 1337270, 1337271, 1337272, 1345593, 1365145, 1366973, ...
    * 1328860 => RESOLVED FIXED in Firefox 54 - Install pref to have data: URIs not inherit the security context
    * 1365166 => RESOLVED FIXED in Firefox 55 - rename `security.data_uri.inherit_security_context` to `security.data_uri.unique_opaque_origin`
  * Blocks: 1296976 - Access Denied - You are not authorized to access bug 1296976

* [1331351](https://bugzilla.mozilla.org/show_bug.cgi?id=1331351) - Consider blocking top level window data: URIs 
  * Depends on: 1357386 => RESOLVED FIXED in Firefox 55 - Gather telemetry for toplevel data: URI loads
  * Duplicates: 1360485 => RESOLVED DUPLICATE of bug 1331351 - Firefox is vulnerable to phishing stored in data URI
    * lists a PoC testsite in OP (http://createcharts.esy.es/data-url.html)

**Comment by earthlng** [comment]
`dom.event.highrestimestamp.enabled` - yes we want true but true is the default value in FF54+ as is unlikely ever gonna change - do we really need to add this to the user.js? We're a bit late to the party with this one xD

**Comment by earthlng** [comment]
```js
// Allow Flash async drawing mode in 64-bit release builds
pref("dom.ipc.plugins.asyncdrawing.enabled", true);
// Force the accelerated direct path for a subset of Flash wmode values
pref("dom.ipc.plugins.forcedirect.enabled", true);
```
Flash stuff - who gives a fuck?!? If people still use that piece of shit that's not my problem.

**Comment by Thorin-Oakenpants** [comment]
> `dom.event.highrestimestamp.enabled` - yes we want true but true is the default value in FF54+ and is unlikely ever gonna change - do we really need to add this to the user.js? We're a bit late to the party with this one xD

TBH, I don't think this does jack shit.  dom.enable_user_timing is removed in 55 ( https://bugzilla.mozilla.org/show_bug.cgi?id=1344669 ) . They plan to reintroduce something similar. The way I see it is this is a pref for testing and migration to the new API ( to monotonic from epoch time). They're both still hi-res (who gives a f*k about the truncation, it's still hi-res enough), They are aware of it in bug https://bugzilla.mozilla.org/show_bug.cgi?id=1217238 and see the comment here ( https://bugzilla.mozilla.org/show_bug.cgi?id=1367313 ) .. and see the comment here (whoops can;t find the ref, but one of the tor uplift guys stated that this pref needs to be replaced at some stage)

`dom.event.highrestimestamp.enabled` is a red herring and we can drop it - class, discuss (who's in this class?)

**Comment by Thorin-Oakenpants** [comment]
earthlng commented 8 days ago
https://github.com/ghacksuserjs/ghacks-user.js/issues/87#issuecomment-304534145
changes in beta11 since the beta1 diff in OP: 4 new prefs, 6 changed prefs

These don't seem to have been merged in with the first master post

**Comment by earthlng** [comment]
no, I wanted to wait for stable to update the first post

**Comment by Thorin-Oakenpants** [comment]
just thumbs up if its ok to add - I'll dump some more soon, otherwise post nits

```js
/* 0361: disable Activity Stream (system addon) (FF54+)
 * [1] https://wiki.mozilla.org/Firefox/Activity_Stream ***/
pref("browser.newtabpage.activity-stream.enabled", false);
```

**Comment by Thorin-Oakenpants** [comment]
```js
/* 0809: disable location bar displaying preloaded entries (FF54+)
 * [1] https://bugzilla.mozilla.org/show_bug.cgi?id=1211726 ***/
user_pref("browser.urlbar.usepreloadedtopurls.enabled", false)
```

**Comment by Thorin-Oakenpants** [comment]
```js
/* 0863: disable Form Autofill (FF54+)
 * [1] https://www.ghacks.net/2017/05/24/firefoxs-new-form-autofill-is-awesome/
 * [2] https://wiki.mozilla.org/Firefox/Features/Form_Autofill ***/
user_pref("browser.formautofill.enabled", false);
```

**Comment by earthlng** [comment]
~~displaying~~ suggesting preloaded ~~entries~~ top websites ? otherwise :thumbsup:

**Comment by earthlng** [comment]
"preloaded" is also not great. pre-seeded is what they use in the bugzilla. IDK

**Comment by Thorin-Oakenpants** [comment]
They actually also use preload as a term, pre-seed is just what the OP called it
- preload: 13 matches - to be fair, a lot are quoted - the pref even uses it
- pre-seed: 7 matches - all stemming from the title and first post

pre-seed seems weird to me, not very user-friendly to non coders. Whereas everyone knows preloading means drinking up large at home before hitting the expensive bars at night-time :)

**Comment by earthlng** [comment]
> `dom.event.highrestimestamp.enabled`
> Edit: If anything we should enforce it as false, so, I assume, we fall back to epoch time, and the old dom.enable_user_timing is still applied (with epoch) at least in 54 until it is removed in 55

afaik setting it to false would make things worse. Look at the TBB ticket you linked again. This pref *does* something but it's not about precision or in-precision. Hopefully they'll bring back `dom.enable_user_timing` - IDK what the interaction is between these 2 prefs.

**Comment by Thorin-Oakenpants** [comment]
I think we can agree to ignore the pref :)

**Comment by earthlng** [comment]
> pre-seed seems weird to me, not very user-friendly to non coders. Whereas everyone knows preloading means drinking up large at home before hitting the expensive bars at night-time :)

lol! yeah I don't really care - preloaded is fine too

**Comment by earthlng** [comment]
FYI no changes in Beta13 prefs since Beta11

**Comment by Thorin-Oakenpants** [comment]
This is tricky. It not really about 1102 which is "web content". It's about WE. See 3rd/4th comments here: https://www.reddit.com/r/firefox/comments/62ybve/will_multiprocess_firefox_going_to_be_a_true/

```js
/* 1106: control number of WebExtension processes ***/
   // user_pref("dom.ipc.processCount.extension", 1);
```

and replace `1102: control number of e10s processes` with `1102: control number of content rendering processes`

**Comment by earthlng** [comment]
> replace `1102: control number of e10s processes` with `1102: control number of content rendering processes`

yes, that's good but 1106 should be 1102b IMO

I'd say drop the 2 ipc.plugins prefs in 1110 (because fuck Flash), move it to 3000 and also add 1111 to 3000 right above it because both items don't fit in e10s anyway and not too many people will bother to enable and look at sandbox logging fe
```js
/* 3030: enable sandbox logging ***/
   // user_pref("security.sandbox.logging.enabled", false);
/* 3031: set sandbox level. DO NOT MEDDLE WITH THESE. ...
   ...  ***/
   // user_pref("security.sandbox.content.level", "donotuse");
```

**Comment by earthlng** [comment]
https://github.com/ghacksuserjs/ghacks-user.js/commit/dd01dd54c65de3c62ee1e91666ed3dedec44b95b
dude! why the rush all of a sudden?! at least let me respond before you make these changes :)

**Comment by Thorin-Oakenpants** [comment]
soz .. am impatient

-  cheers, fixed the false->true before the commit
- 1106->1102b - I want to get away from letters. The order is slightly wonky, but it's very small section. More prefs are due to come (and go), we can always revisit it
- drop the two ipc.plugins, leave em there for info for now. We only *just* added this section
- e10s and sandboxing belong together, esp as sandboxing doesn't warrant a whole section etc. They're all about security - processes for gui-core, content, extensions, gpu, stacks, whatnots, per tab, etc, and sandboxing is a natural progression from that.
- move stuff to 3000s - i want to get rid of the 3000's as they do not fit in the scope of this project and move them to a new repo under the ghacks organization (maybe next year). I get that logging is outside the scope (privacy/security/anti-FP etc), but it logically needs to be next to the actual sandboxing pref. Sandboxing *is* in our scope.

I'm happy with the commit

**Comment by Thorin-Oakenpants** [comment]
should we add `media.peerconnection.ice.tcp`=>false to 2001 - [1176382](https://bugzilla.mozilla.org/show_bug.cgi?id=1176382)

**Comment by earthlng** [comment]
`browser.migrate.automigrate.enabled` - the change probably won't land in the FF54 Release (see beta11 diff) and I will remove it from the list tomorrow if that's the case
`gfx.downloadable_fonts.otl_validation` - I already commented on that [here](https://github.com/ghacksuserjs/ghacks-user.js/issues/87#issuecomment-304834100)
`extensions.screenshots.system-disabled` - do we want to wait till this gets set to false by default, or ignore it completely?

**Comment by Thorin-Oakenpants** [comment]
@earthlng
- `browser.migrate.automigrate.enabled` - I'm pretty sure this will only happens to new profiles (yes, shitty behavior to import and then ask for forgiveness), and by the time someone sets this to false it's too late. I'm also not sure if this set to false acts like a master switch and then disables the ability to roll back. It's part of the "onboarding" experience .. i.e why did a new FF user come on board? I guess we'll know more about it as we get closer to 57.
- `gfx.downloadable_fonts.otl_validation` - I read the info/links you provided ages ago and revisited them, but couldn't decide what to do. I think we might ignore it for now.
- `extensions.screenshots.system-disabled` - I like this, I hope it gets a few more tweaks.  We could add it to the personal section. Users do not have to upload to the cloud which I think requires an account, and (for me) it means one less add-on come 57. I use screengrab fix version. It allows a set directory for saving as png or jpg, and you can grab selection/whole-page/visible-page as image or copy to clipboard - it's the only decent screen grabber I've found - tried a few others (WE) but they really suck. Would much prefer an inbuilt Mozilla solution for peace of mind too.

**Comment by Thorin-Oakenpants** [comment]
:+1:  or nits

```js
/* 3029: disable Firefox Screenshots (FF54+)
 * [1] https://www.ghacks.net/2017/05/28/firefox-screenshots-integrated-in-firefox-nightly/
 * [2] https://github.com/mozilla-services/screenshots ***/
   // user_pref("extensions.screenshots.system-disabled", true);
```

**Comment by Thorin-Oakenpants** [comment]
Well, I've upgraded to 54 and `browser.migrate.automigrate.enabled` is still set as false by default. So can that be removed from the diffs? Can we merge the diffs above into the first post? Are we ready to close this and release an alpha?

**Comment by earthlng** [comment]
```js
// This is referred only when both "intl.tsf.enable" and "intl.tsf.support_imm"
// are true.  When this is true, default IMC is associated with focused window
// only when active keyboard layout is a legacy IMM-IME.
pref("intl.tsf.associate_imc_only_when_imm_ime_is_active", false);
// Whether default IMC should be associated with focused window when MS-IME
// for Japanese on Win10 is active.  MS-IME for Japanese on Win10 has a crash
// bug.  While restoring default IMC when MS-IME for Japanese is active,
// it sometimes crashes after Creators Update.  This pref avoid the crash.
pref("intl.tsf.hack.ms_japanese_ime.do_not_associate_imc_on_win10", true);
```
If you agree that we can ignore these 2 then we can close this and you can release an alpha.

**Comment by Thorin-Oakenpants** [comment]
> If you agree that we can ignore these 2 then we can close this and you can release an alpha.

I agree

**Comment by Thorin-Oakenpants** [comment]
[earthlng user.js v54.0-alpha](https://github.com/ghacksuserjs/ghacks-user.js/releases/tag/v54.0-alpha)

**Comment by earthlng** [comment]
Thx I'll start on the changelog right away - boy there are quite a few pref changes this time though :(

**Comment by Thorin-Oakenpants** [comment]
Don't worry about the issue links, we edit those in afterwards
